routes/conversation.js

```diff
@@ -169,6 +169,8 @@ exports.leave = function (req, res){
 			var participantIds = convo.participants.map(function (p){return p._id});
 			var userString = JSON.stringify(req.user._id);
 			var participantIdStrings = participantIds.map(function (id){return JSON.stringify(id)});
+			// Should this 'found' variable definition be changed to the equivalent 'foundIndex' variable
+			// definitions used below in 'exports.archive'?
```

Comment:
Wow, that's definitely correct.  I think you've uncovered a bug!  The leave function should be using a foundIndex of sorts when splicing.  You're right, as it is now, it does seem like it would always remove the first entry, which is not what we want.  Let's change it!